sdk/textanalytics/ai-text-analytics/src/lro/analyze/operation.ts

```diff
@@ -0,0 +1,384 @@
+    private options: BeginAnalyzeOptions = {},
+    private statusOptions: TextAnalyticsStatusOperationOptions
+  ) {
+    super(state);
+  }
+
+  /**
+   * should be called only after all the status of the analyze jobs became
+   * "succeeded" and it returns an iterator for the results and provides a
+   * byPage method to return the results paginated.
+   */
+  private listAnalyzeResultsByPage(
+    jobId: string,
+    options: AnalyzeJobStatusOptions = {}
+  ): PagedAsyncIterableAnalyzeResults {
+    const iter = this._listAnalyzeResults(jobId, options);
+    const pagedIterator = {
+      next() {
+        return iter.next();
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      byPage: (settings?: PageSettings) => {
+        const pageOptions = { ...options, top: settings?.maxPageSize };
+        return this._listAnalyzeResultsPaginated(jobId, pageOptions);
+      }
+    };
+    return pagedIterator;
+  }
+
+  /**
+   * returns an iterator to the results of an analyze job.
+   */
+  private async *_listAnalyzeResults(
+    jobId: string,
+    options?: AnalyzeJobStatusOptions
+  ): AsyncIterableIterator<AnalyzeResult> {
+    for await (const page of this._listAnalyzeResultsPaginated(jobId, options)) {
+      yield* page;
+    }
+  }
+
+  /**
+   * returns an iterator to arrays of the results of an analyze job.
+   */
+  private async *_listAnalyzeResultsPaginated(
+    jobId: string,
+    options?: AnalyzeJobStatusOptions
+  ): AsyncIterableIterator<AnalyzeResultsArray> {
+    let response = await this._listAnalyzeResultsSinglePage(jobId, options);
+    yield response.result;
+    while (response.skip) {
+      const optionsWithContinuation: AnalyzeJobStatusOptions = {
+        ...options,
+        top: response.top,
+        skip: response.skip
+      };
+      response = await this._listAnalyzeResultsSinglePage(jobId, optionsWithContinuation);
+      yield response.result;
+    }
+  }
+
+  /**
+   * returns an iterator to arrays of the sorted results of an analyze job.
+   */
+  private async _listAnalyzeResultsSinglePage(
+    jobId: string,
+    options?: AnalyzeJobStatusOptions
+  ): Promise<AnalyzeResultsWithPagination> {
+    const { span, updatedOptions: finalOptions } = createSpan(
+      "TextAnalyticsClient-_listAnalyzeResultsSinglePage",
+      options || {}
+    );
+    try {
+      const response = await this.client.analyzeStatus(
+        jobId,
+        operationOptionsToRequestOptionsBase(finalOptions)
+      );
+      const result: AnalyzeResultsArray = [
+        ...makeAnalyzeEntitiesResultArray(this.documents, response.tasks.entityRecognitionTasks),
+        ...makeAnalyzePiiEntitiesResultArray(
+          this.documents,
+          response.tasks.entityRecognitionPiiTasks
+        ),
+        ...makeAnalyzeKeyPhrasesResultArray(this.documents, response.tasks.keyPhraseExtractionTasks)
+      ];
+      return response.nextLink
+        ? { result, ...nextLinkToTopAndSkip(response.nextLink) }
```

Comment:
if the request contained `showStats`, will that be passed in the next page request?

Reply:
yup, only `top` and `skip` are modified and the rest of the options stay the same.